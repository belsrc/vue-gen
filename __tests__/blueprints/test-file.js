const genTest = require('./../../src/blueprints/scss-file');

const compName = 'TestComponent';
const fileName = 'test-component';

describe('genTest', () => {
  test('has the expected ouput structure', () => {
    const actual = genTest(compName, fileName);

    expect(actual).toMatchSnapshot();
  });
});
