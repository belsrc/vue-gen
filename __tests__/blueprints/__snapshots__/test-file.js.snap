// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genTest has the expected ouput structure 1`] = `
"import '@babel/register';
import { shallowMount } from '@vue/test-utils';
import TestComponent from './test-component';

// Template uses Jest and Vue test utils as an example
// npm i -D jest @vue/test-utils
// But your free to use whatever test setup you prefer

describe('TestComponent', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.clearAllMocks();
  });

  describe('Computed', () => {
    // test('should have correct <computed-name> computed value', () => {
    //   const wrapper = shallowMount(TestComponent);
    //
    //   const actual = wrapper.vm.<computed-name>;
    //
    //   expect(actual).toEqual(<value>);
    // });
    test.skip('NOT IMPLEMENTED', () => {});
  });

  describe('Methods', () => {
    // test('<method-name> should return correct value', () => {
    //   const wrapper = shallowMount(TestComponent);
    //
    //   const actual = wrapper.vm.<method-name>();
    //
    //   expect(actual).toEqual(<value>);
    // });
    test.skip('NOT IMPLEMENTED', () => {});
  });

  describe('Rendering', () => {
    // test('should not show some selector', () => {
    //   const wrapper = shallowMount(TestComponent);
    //
    //   const actual = wrapper.find('.class-selector).exists();
    //
    //   expect(actual).not.toBeTruthy();
    // });
    test.skip('NOT IMPLEMENTED', () => {});
  });
});
"
`;

exports[`genTest has the expected ouput structure when functional 1`] = `
"import '@babel/register';
import { shallowMount } from '@vue/test-utils';
import TestComponent from './test-component';

// Template uses Jest and Vue test utils as an example
// npm i -D jest @vue/test-utils
// But your free to use whatever test setup you prefer

describe('TestComponent', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.clearAllMocks();
  });

  describe('Rendering', () => {
    // test('should not show some selector', () => {
    //   const wrapper = shallowMount(TestComponent);
    //
    //   const actual = wrapper.find('.class-selector).exists();
    //
    //   expect(actual).not.toBeTruthy();
    // });
    test.skip('NOT IMPLEMENTED', () => {});
  });
});
"
`;
